python-checks/src/main/java/org/sonar/python/checks/BuiltinGenericsOverTypingModuleCheck.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * SonarQube Python Plugin
+ * Copyright (C) 2011-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.python.checks;
+
+import java.util.Map;
+import java.util.Optional;
+import javax.annotation.Nullable;
+import org.sonar.check.Rule;
+import org.sonar.plugins.python.api.PythonSubscriptionCheck;
+import org.sonar.plugins.python.api.SubscriptionContext;
+import org.sonar.plugins.python.api.symbols.Symbol;
+import org.sonar.plugins.python.api.tree.Expression;
+import org.sonar.plugins.python.api.tree.HasSymbol;
+import org.sonar.plugins.python.api.tree.SubscriptionExpression;
+import org.sonar.plugins.python.api.tree.Tree;
+import org.sonar.plugins.python.api.tree.TypeAnnotation;
+
+import static java.util.Map.entry;
+
+@Rule(key = "S6545")
+public class BuiltinGenericsOverTypingModuleCheck extends PythonSubscriptionCheck {
+
+  private static final String MESSAGE = "Use the built-in generic type: `%s`, instead of its typing counterpart.";
```

Comment:
_super minor, subjective_ I think we can phrase it without the `:`:
```suggestion
  private static final String MESSAGE = "Use the built-in generic type `%s` instead of its typing counterpart.";
```

Feel free to ignore this if it doesn't make sense to you.

Reply:
Sure I think it is clear enough with the back ticks only